.github/workflows/my-ci-pipeline.yml

```diff
@@ -0,0 +1,19 @@
+name: Continuous Integration
+on: 
+  push:
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+jobs:
+ build:
+  name: Build and test
+  runs-on: ubuntu-latest
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Build
+      run: docker build --target development --tag todo_app:test .
+    - name: Test
+      run: docker run --env-file ./.env.test --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app/todo_app --mount type=bind,source="$(pwd)"/tests,target=/todo_app/tests --entrypoint poetry todo_app:test run pytest
```

Comment:
This works, but it might be nicer to add an explicit stage to your Dockerfile for this; e.g. one that:
* Copies the test files in
* Uses the test command as the entrypoint
That way this command doesn't need to be quite as intricate, and we can drop the bind mount since the files are already present